src/components/header/Header.test.tsx

```diff
@@ -33,4 +34,20 @@ describe('Header', () => {
     const shopTitle = screen.getByText('Shoply');
     expect(shopTitle).toBeInTheDocument();
   });
+
+  it('Displays the SearchInput component when the user clicks the search button, then removes it when clicked again ', async () => {
+    render(<Header />);
+
+    const searchInput = screen.queryByRole('textbox');
+    expect(searchInput).not.toBeInTheDocument();
+
+    const searchIcon = screen.getByTestId('magnifying-glass');
+    userEvent.click(searchIcon);
+    const searchInputDisplayed = await screen.findByRole('textbox');
+    expect(searchInputDisplayed).toBeInTheDocument();
```

Comment:
You could use `toBeVisible` here to get closer to how we want a user to experience the component